engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -2508,6 +2509,18 @@ double paladin_t::resource_loss( resource_e resource_type, double amount, gain_t
   {
     buffs.blessing_of_dusk -> trigger();
   }
+
+  if (
+      resource_type == RESOURCE_HEALTH &&
+      legendary.reign_of_endless_kings -> ok() &&
+      ! buffs.reign_of_ancient_kings -> up() &&
+      health_percentage()/100 < legendary.reign_of_endless_kings -> effectN( 2 ).percent() &&
+      health_percentage()/100 + amount / max_health() >= legendary.reign_of_endless_kings -> effectN( 2 ).percent()
+    )
+  {
+    buffs.guardian_of_ancient_kings -> trigger( legendary.reign_of_endless_kings -> effectN( 2 ).trigger() -> duration() );
+    buffs.reign_of_ancient_kings -> trigger();
+  }
```

Comment:
This seems a bit circumvoluted... 

So the spelldata seems to indicate that the player gains the goak buff (ID 337851, with damage reduction data in the original goak buff ID 86657). Which one is on the player ingame?
You're creating a new buff called "reign of ancient kings" solely to handle the visible icd on the player, it might be better for simplicity and report clarity to make it the actual damage reduction buff (especially if it's the one used ingame) and just give it a cooldown duration equal to the one that is stored in 337852.

Last, did you make sure that the proc doesn't happen if the player is already below the threshold when taking damage (assuming the internal cooldown is up).

Reply:
337851 is a 4sec buff that shows up in the log but not on the player's buffs, so I don't know what it's even doing. 337852 is the debuff that prevents it proccing for 45sec. 86659 is the actual goak buff itself you see on your character.
In-game and in the sim it won't proc if you're already below the threshhold when the debuff expires until you heal back over it.